### ISSUE TITLE
New Rule: no-mergable-namespace

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#843]
Similar to scss-lint's [MergeableSelector rule](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#mergeableselector), we should lint against mergeable namespaces within the same file:

```ts
namespace Lint {
    ...
}

...

namespace Lint { // failure
    ...
}
```
